hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/bootstrap/HoodieSparkBootstrapSchemaProvider.java

```diff
@@ -71,11 +72,20 @@ protected Schema getBootstrapSourceSchema(HoodieEngineContext context, List<Pair
   }
 
   private static Schema getBootstrapSourceSchemaParquet(HoodieWriteConfig writeConfig, HoodieEngineContext context, Path filePath) {
-    MessageType parquetSchema = new ParquetUtils().readSchema(context.getHadoopConf().get(), filePath);
+    Configuration hadoopConf = context.getHadoopConf().get();
+    MessageType parquetSchema = new ParquetUtils().readSchema(hadoopConf, filePath);
+
+    hadoopConf.set(
+        SQLConf.PARQUET_BINARY_AS_STRING().key(),
+        SQLConf.PARQUET_BINARY_AS_STRING().defaultValueString());
+    hadoopConf.set(
+        SQLConf.PARQUET_INT96_AS_TIMESTAMP().key(),
+        SQLConf.PARQUET_INT96_AS_TIMESTAMP().defaultValueString());
+    hadoopConf.set(
+        SQLConf.CASE_SENSITIVE().key(),
+        SQLConf.CASE_SENSITIVE().defaultValueString());
+    ParquetToSparkSchemaConverter converter = new ParquetToSparkSchemaConverter(hadoopConf);
```

Comment:
[SPARK-36935](https://github.com/apache/spark/pull/34199) Constructor is changed for `ParquetToSparkSchemaConverter` in Spark 3.3. Switched to use another constructor that takes conf for bwc

Reply:
> [SPARK-36935](https://github.com/apache/spark/pull/34199) Constructor is changed for `ParquetToSparkSchemaConverter` in Spark 3.3. Switched to use another constructor that takes conf for bwc

Is the version of spark 3.3 also compatible with this?

Reply:
Yes, this is calling a constructor shared between all spark versions

Reply:
I'm ok with using adapter. but please correct me, this is a java class, and I don't think we can implement `SparkAdapterSupport` without rewriting its methods again here

Reply:
Even in scala `ParquetToSparkSchemaConverter` defined default values, but those default values can be referred to when it's instantiated in Java. Using the constructor that takes conf obj is good enough here

Reference: https://github.com/scala/bug/issues/4278